SpatialGDK/Source/SpatialGDKEditor/Private/SpatialTestSettings.cpp

```diff
@@ -0,0 +1,103 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+#include "SpatialTestSettings.h"
+
+FSpatialTestSettings::FSpatialTestSettings()
+	: OriginalSpatialGDKSettings(nullptr)
+	, OriginalLevelEditorPlaySettings(nullptr)
+	, OriginalSpatialGDKEditorSettings(nullptr)
+	, OriginalGeneralProjectSettings(nullptr)
+	, OriginalEditorPerformanceSettings(nullptr)
+{
+}
+
+void FSpatialTestSettings::Override(const FString& MapName)
+{
+	// Back up the existing settings so they can be reverted later
+	Backup();
+	// First override the settings from the base config file, if it exists
+	Load(BaseOverridesFilename);
+	// Then override the settings from the map specific config file, if it exists
+	FString MapOverridesFilename = OverrideSettingsBaseFilename + FPackageName::GetShortName(MapName) + (OverrideSettingsFileExtension);
+	Load(MapOverridesFilename);
+}
+
+void FSpatialTestSettings::Backup()
+{
+	// Duplicate original settings but use different outer - if same outer is reused the object is not duplicated and pointer is to the same
+	// Object Having the same name causes runtime exceptions
+	// Use additional object flag RF_Standalone to avoid early GC - destroy when restoring settings
+	OriginalSpatialGDKSettings = NewObject<USpatialGDKSettings>(
+		GetTransientPackage(), USpatialGDKSettings::StaticClass(), NAME_None,
+		EObjectFlags(RF_Public | RF_ClassDefaultObject | RF_ArchetypeObject | RF_Standalone), GetMutableDefault<USpatialGDKSettings>());
```

Comment:
```suggestion
		EObjectFlags(RF_Public | RF_ClassDefaultObject | RF_ArchetypeObject | RF_Standalone), GetDefault<USpatialGDKSettings>());
```
Nit: I *think* this should work as well, as we are only reading them? But not 100% sure

Reply:
Good point. But I've checked and the NewObject function requires a UObject* and does not work with a const UObject*.